soepy/test/test_unit.py

```diff
@@ -143,4 +146,85 @@ def test4():
         np.testing.assert_array_equal(df.shape[0], constr["AGENTS"] * constr["PERIODS"])
 
 
+def test5():
+    """This test ensures that the state space creation generates the correct admissible
+    state space points for the first 4 periods."""
+
+    model_params = namedtuple("model_params", "num_periods educ_range educ_min")
+    model_params = model_params(4, 3, 10)
+
+    states_all, states_number_period, _, _ = pyth_create_state_space(model_params)
+
+    # Control for correct number of states in each period.
+    np.testing.assert_array_equal(states_number_period, [1, 4, 13, 30])
+
+    # Control that the states are correct
+    states_true = np.full((576, 4), MISSING_INT)
```

Comment:
I suggest creating one hard-coded `states_true` container that has exactly the same structure as `states_all` and then we can just run one assert statement on the two containers without the need to create the temporary `states_period1` etc.